### PR TITLE
Rewrite get-date pipeline step to remove warning

### LINKED ISSUE #118: Pipeline step "Get Date" produces warning
In our pipeline-runs, the step `get-date` produces a warning:

```
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

See e.g. [here](https://github.com/quarkiverse/quarkus-artemis/actions/runs/3636261517/jobs/6136072642#step:4:7). 

We should rewrite this part to remove the warning.

### PR DESCRIPTION
**Pull request by turing85**
- Fixes #118

### CODE REVIEW

**Comment by gastaldi** [comment]
I'd be in favor of removing this step and use `cache: 'maven`` in the `actions/setup-java@v3` step. See https://github.com/quarkiverse/quarkus-prettytime/blob/main/.github/workflows/build.yml

**Comment by turing85** [comment]
> I'd be in favor of removing this step and use `cache: 'maven'` in the `actions/setup-java@v3` step. See https://github.com/quarkiverse/quarkus-prettytime/blob/main/.github/workflows/build.yml

Do we have an automatic way to prevent the cache from growing over time, or is this just a no-concern?

**Comment by gastaldi** [comment]
The cache is invalidated when the pom.xml changes, so that shouldn't be a problem after a release

**Review by gastaldi — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,23 +31,11 @@ jobs:
       - uses: actions/checkout@v2
```
Comment (gastaldi) [inline .github/workflows/build.yml:31]:
```suggestion
      - uses: actions/checkout@v3
```

Please bump `actions/checkout` too

**Pull request merged**